### PR TITLE
Improve kinesis posting logic

### PR DESCRIPTION
**Pull request by jmathewt**
Fixed bug. Please check the logic esp in the ResourceIdProcessor class.

### CODE REVIEW

**Comment by nonword** [comment]
Looks like I need some env vars. I gather I need these?

```
export resourceType=
export redisHost=
export redisPort=
export kinesisResourceUpdateStream=
export kinesisResourceRetrievalRequestStream=
export sierraApi=
export accessTokenUri=
export clientId=
export clientSecret=
export grantType=
export pollDelay=
```

Are sensible defaults documented somewhere?

**Comment by jmathewt** [comment]
Hey Paul,
I have to make some changes to documentation to reflect some of these
changes.
For now, you can use the following -

 export resourceType=bibs|items
export redisHost=localhost (you will need the redis server running locally)

export redisPort=6379 is the default port
export kinesisResourceUpdateStream=SierraBibUpdate|SierraItemUpdate(depending
on whether your resource type is bibs|items)
export kinesisResourceRetrievalRequestStream=SierraBibRetrievalRequest|SierraItemRetrievalRequest

export sierraApi=<will email this to you>
export accessTokenUri=<will email this to you>

export clientId=<will email this to you>

export clientSecret=<will email this to you>

export grantType=client_credentials
export pollDelay=30s


In addition to this, your amazon keys have to be set in one of the
following ways -
http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html.
Setting them as env variables will be easier (AWS_ACCESS_KEY_ID =
<your_aws_key> and AWS_SECRET_ACCESS_KEY = <your_aws_secret>)

**Comment by kfriedman** [comment]
Jobin, I took a look but it's a little hard to digest. Do you think you can setup a time on Monday to walk through the logic and do a real-time code review? That might be easier/more productive. Thanks!

**Comment by nonword** [comment]
Great discussion the other day. I think most of my concerns were addressed. I've worried a little about system time though. If you're building 5min fetch ranges using local system time, you're going to get 0 results if your local system time is ahead of sierra system time by 5 or more minutes. (You'll keep asking for records updated in the future as far as sierra is concerned.) It's the kind of thing shouldn't happen, so it probably will. Any way to address that? By checking something in api response to confirm that system time offset is less than 5 mins?

**Comment by jmathewt** [comment]
Created a new issue to address @nonword brought up.

File: README.md [hunk 1]
```diff
@@ -1,10 +1,40 @@
 # SierraUpdatePoller
```
Comment (kfriedman) [inline README.md:1]: Can you maybe add deployment instructions? Maybe next version if it's too much work.

**Comment by kfriedman** [comment]
I did my best to review and, other than the time difference issue that @nonword brought up and we discussed, the logic seems sound to me. Nice work!

File: README.md (hunk 1 above)
Comment (jmathewt) [reply README.md:1]: @kfriedman,

I added some description in the documentation will that suffice? What do you mean by deployment instructions?

**Pull request merged**